File: src/main/java/infrastructure/finnhub/dto/FinnhubQuoteDto.java
package infrastructure.finnhub.dto;

import domain.stock.Quote;

//===========================================================//
/**
 *  Data Transfer Object for Finnhub quote response.
 *  <p>
 *     This record represents the raw response format returned by Finnhub.
 *  </p>
 * @param c current price
 * @param d change
 * @param dp percent change
 * @param h high price of the day
 * @param l low price of the day
 * @param o open price of the day
 * @param pc previous close price
 */
//===========================================================//

public record FinnhubQuoteDto(
  double c,
  double d,
  double dp,
  double h,
  double l,
  double o,
  double pc
){
   public Quote toDomain(final String symbol){
      return new Quote(
        symbol,
        c,
        d,
        dp,
        h,
        l,
        o,
        pc,
        System.currentTimeMillis()
      );
   }
}

File: src/main/kotlin/infrastructure/network/finnhub/dto/FinnhubQuoteDto.kt
package infrastructure.network.finnhub.dto

import domain.assets.Quote

//===========================================================//
/**
 * Data Transfer Object for Finnhub quote response.
 *
 *
 * This record represents the raw response format returned by Finnhub.
 *
 * @param c current price
 * @param d change
 * @param dp percent change
 * @param h high price of the day
 * @param l low price of the day
 * @param o open price of the day
 * @param pc previous close price
 */
//===========================================================//

data class FinnhubQuoteDto(
    val c: Double,
    val d: Double,
    val dp: Double,
    val h: Double,
    val l: Double,
    val o: Double,
    val pc: Double
) {
    fun toDomain(): Quote {
        return Quote(
            c,
        )
    }
}